pxf/pxf-service/src/main/java/org/apache/hawq/pxf/service/rest/InvalidPathResource.java

```diff
@@ -153,9 +167,39 @@ private String parseVersion(String path) {
     }
 
     /**
+     * Parses the version part from the path.
```

Comment:
1. Please update doc - it parses the part after the version part, right? /pxf/version/endpoint
2. could you please refactor parseVersion() to also use PathSegment? 
Actually, both functions can perhaps be consolidated. [e.g. parsePathSegment(path, VERSION), parsePath(path, ENDPOINT)]